### PR TITLE
chore(importer): refactor strategies to return single metadata instruction

### PR DESCRIPTION
**Pull request by JeroenVinke**

### CODE REVIEW

File: lib/importer/strategies/browser-section.js [hunk 1]
```diff
@@ -23,22 +22,19 @@ let BrowserSectionStrategy = class {
...
-        main: path.posix.basename(moduleId)
-      };
-
-      logger.info(`Adding the following configuration to the '${targetBundle.name}' bundle`, configuration);
-      return this.patcher.addOrReplaceDependency(targetBundle, configuration);
-    })
-    .then(() => this.patcher.writeAureliaJSON());
+    let browser = this.package.packageJSON.browser;
+    let moduleId = this.package.calculateModuleId(browser);
+    let targetBundle = this.patcher.getDefaultVendorBundle();
```
Comment (Thanood) [inline lib/importer/strategies/browser-section.js]: I think the `targetBundle` is not needed anymore. 😃

File: lib/importer/services/tutorial.js [hunk 2]
```diff
@@ -10,17 +10,26 @@ module.exports = class {
...
-  start() {
+  start(metadata) {
     return this.getTutorial()
     .then(tutorial => {
-      if (!tutorial) {
-        logger.debug(`There is no tutorial for "${this.package.name}" (version ${this.package.version})`);
-        return;
+      logger.info('*********** Tutorial ***********');
+
+      if (tutorial) {
```
Comment (Thanood) [inline lib/importer/services/tutorial.js]: Just an idea: If there's no tutorial we could hint users/maintainers that it's possible to add a tutorial to package.json.
Now that the "there is no tutorial" message is gone, there is also no hint anymore.

**Comment by adriatic** [comment]
I will wait for Jeroen's response before adding my own comments

File: lib/importer/services/tutorial.js (hunk 2 above)
Comment (JeroenVinke) [inline lib/importer/services/tutorial.js]: Good one!

**Comment by JeroenVinke** [comment]
> Looks good to me, just a nitpick and a question/idea. 😃

> One thing that I couldn't grasp, though. If the package provides for example a JSPM section, the JSPM strategy is executed. import-engine then fulfills the promise and the whole strategy "chain" is finished, right?

Yes

> Resource detection/inclusion is done in amodro-strategy. If my above statement is correct then there won't be any resource detection if a JSPM config is found.

Correct. I have updated the PR. But i'd like to see if it's possible to have a single class responsible for putting together a configuration (using a builder pattern), so that's what i'm going to try now

**Comment by Thanood** [comment]
Since you've added `resourceInclusion` to all relevant strategies. Would it be beneficial (in terms of "chain of responsibility") to put this in `MetadataService.execute`?

**Comment by JeroenVinke** [comment]
@Thanood I am done with refactoring, can you take a look? 

- Strategies now modify the [package](https://github.com/monterey-framework/cli/blob/metadata/lib/importer/package.js) and the package exports its configuration. 
- [Promise.resolve()](https://github.com/monterey-framework/cli/blob/metadata/lib/importer/package-importer.js#L39) is used, so strategies can **optionally** return a promise

This has made the code much cleaner in my opinion, so i'd like to hear what you think 😄

File: lib/importer/services/tutorial.js [hunk 3]
```diff
@@ -42,7 +42,13 @@ module.exports = class {
...
         }
       }
-
+      
+      logger.info('*********** Importing the module ***********');
+      logger.info('The following import statements are possible:');
+      logger.info(`import '${this.package.name}';`);
+      logger.info(`import ${this.package.name} from '${this.package.name}';`);
+      logger.info(`import * as ${this.package.name} from '${this.package.name}';`);
+      logger.info('We are looking into ways to detect what is the right one');
```
Comment (Thanood) [inline lib/importer/services/tutorial.js]: I really like that. 😃

**Comment by Thanood** [comment]
I like these changes better than my idea ("Would it be beneficial ... to put this in MetadataService.execute") now that I realize there are strategies which actually don't lookup resources (f.i. aurelia-registry).

**Pull request merged**